app/scripts/directives/blockRegistration.js

```diff
@@ -77,24 +77,20 @@ angular
               registrantIndex
             ].answers.push($scope.answer);
 
-          $scope.$watch(
-            'answer',
-            function(answer, oldAnswer) {
-              if (
-                angular.isUndefined(answer) ||
-                angular.isUndefined(oldAnswer) ||
-                angular.equals(answer, oldAnswer)
-              ) {
-                return;
-              }
+          $scope.$watchCollection('answer', function(answer, oldAnswer) {
+            if (
+              angular.isUndefined(answer) ||
+              angular.isUndefined(oldAnswer) ||
+              angular.equals(answer, oldAnswer)
+            ) {
+              return;
+            }
 
-              RegistrationCache.updateCurrent(
-                $scope.conference.id,
-                $scope.currentRegistration,
-              );
-            },
-            true,
```

Comment:
Have you tested this? Removing this disables deep equality checks right? If we don't need it, that's great, just seems like it might miss updates if `answer` is an object or array.

Reply:
I read an article about improving angularJS performance and this was a suggestion. Everything seems to run fine but no performance boost was gained so I will switch it back just in case.